lib/nerves_network/dhcp_manager.ex

```diff
@@ -392,13 +392,19 @@ defmodule Nerves.Network.DHCPManager do
   end
 
   defp start_link_local(state) do
-    {:ok, ifsettings} = Nerves.NetworkInterface.status(state.ifname)
-    ip = generate_link_local(ifsettings.mac_address)
+    {:ok, ifstatus} = Nerves.NetworkInterface.status(state.ifname)
+    ip = generate_link_local(ifstatus.mac_address)
 
     scope(state.ifname)
     |> SystemRegistry.update(%{ipv4_address: ip})
 
     :ok = Nerves.NetworkInterface.setup(state.ifname, ipv4_address: ip)
+
+    {:ok, settings} = Nerves.NetworkInterface.settings(state.ifname)
```

Comment:
Do we want this to be as volatile as that, having an exact pattern match? Or would a case statement along with an error message be more appropriate?

Just asking.

Reply:
Yeah , I was thinking about it already. At this stage if the network interface is not there in the middle of DHCP negotiation that started for that interface it would probably be a disaster in any way. May happen with the hot-pluggable USB interfaces but that would be another user-scenario I deem.

Reply:
actually let me prepare the code better for the future...